### PR TITLE
Modify `isElementInScope` check to disable `FocusScope` with another `FocusScope` as child

### PR DESCRIPTION
**Pull request by Rafal2228**
Closes https://github.com/adobe/react-spectrum/issues/2120

## ✅ Pull Request Checklist:

- [X] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [X] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [X] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Originally reported [issue](https://codesandbox.io/s/sharp-fast-8q7wj?file=/src/App.js), should now work regardless of `FocusScope` being mounted inside `Portal` or not

### CODE REVIEW

File: packages/@react-aria/focus/src/FocusScope.tsx [hunk 1]
```diff
@@ -273,8 +273,25 @@ function isElementInAnyScope(element: Element, scopes: Set<RefObject<HTMLElement
...
+  'data-focus-scope-start',
+  'data-focus-scope-end'
+];
+
+function isFocusScopeDirectChild(scopes: HTMLElement[]) {
+  return scopes.some(scope => focusScopeDataAttrNames.some(name => scope.getAttribute(name) !== null));
+}
+
+function isFocusScopeNestedChild(element: Element) {
+  return focusScopeDataAttrNames.some(name => element.querySelector(`[${name}]`));
```
Comment (snowystinger) [inline packages/@react-aria/focus/src/FocusScope.tsx:286]: lets have the two signatures match
```suggestion
function isFocusScopeNestedChild(scopes: Element[]) {
  return scopes.some(scope => focusScopeDataAttrNames.some(name => element.querySelector(`[${name}]`)));
```

Comment (snowystinger) [reply packages/@react-aria/focus/src/FocusScope.tsx:286]: or the other way around, then line 290 could be like this instead
```
return scopes.some(scope => isFocusScopeDirectChild(scope) || isFocusScopeNestedChild(scope));
```

**Review by snowystinger — COMMENTED** [review]
This seems reasonable to me. Good idea on the span sentinels to mark the scope and thanks for adding tests. One minor change, and I'll test it out when I have some more time.
Would you mind adding a couple stories that illustrate this? you'd need to add the stories directory to this package

File: packages/@react-aria/focus/src/FocusScope.tsx (hunk 1 above)
Comment (Rafal2228) [reply packages/@react-aria/focus/src/FocusScope.tsx]: https://github.com/adobe/react-spectrum/pull/2139/commits/f3e9704b804c306a2d8a293c4348349f4763369e

**Comment by Rafal2228** [comment]
> Would you mind adding a couple stories that illustrate this? you'd need to add the stories directory to this package

Sure thing :) - https://github.com/adobe/react-spectrum/pull/2139/commits/8dc813cff50fe6f1127a64b945c7d096cd00a8bd

**Review by snowystinger — DISMISSED** [review]
Looks good to me in Chrome/Safari/FF

One minor note and looks like it's pre-existing, but if you use the portal option and open 2 "dialogs", you can click on an input in the first "dialog" and get locked into the outer focus scope of that first "dialog"

File: packages/@react-aria/focus/stories/FocusScope.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
```
Comment (solimant) [inline packages/@react-aria/focus/stories/FocusScope.stories.tsx:2]:
```suggestion
 * Copyright 2021 Adobe. All rights reserved.
```

File: packages/@react-aria/focus/stories/FocusScope.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+import {boolean, withKnobs} from '@storybook/addon-knobs';
+import {FocusScope} from '../';
+import React, {ReactNode, useState} from 'react';
+import ReactDOM from 'react-dom';
+import {storiesOf} from '@storybook/react';
+
+const dialogsRoot = 'dialogsRoot';
+
+storiesOf('FocusScope', module)
```
Comment (solimant) [inline packages/@react-aria/focus/stories/FocusScope.stories.tsx:21]: I think we're moving towards Component Story Format (CSF) for stories.

**Comment by Rafal2228** [comment]
> One minor note and looks like it's pre-existing, but if you use the portal option and open 2 "dialogs", you can click on an input in the first "dialog" and get locked into the outer focus scope of that first "dialog"

@snowystinger That's true, but on the other hand in a production ready `Dialog` you would probably didn't allow for clicking on things below it. I would imagine that in such a case the child `Dialog` should have been closed (removed from DOM essentially), so the "locking" inside of parent `FocusScope` seems correct. WDYT?

File: packages/@react-aria/focus/stories/FocusScope.stories.tsx (hunk 3 above)
Comment (Rafal2228) [reply packages/@react-aria/focus/stories/FocusScope.stories.tsx]: https://github.com/adobe/react-spectrum/pull/2139/commits/9ad5946a1c55da57f059a2b9d0134c61d4ab0b3d

**Review by snowystinger — APPROVED** [review]
> @snowystinger That's true, but on the other hand in a production ready Dialog you would probably didn't allow for clicking on things below it. I would imagine that in such a case the child Dialog should have been closed (removed from DOM essentially), so the "locking" inside of parent FocusScope seems correct. WDYT?

Yes, that is correct, I don't know how common a scenario it'd actually be to have nested FocusScopes that aren't nested overlays of some kind. That's why I'm not concerned about fixing it here. More of a note. Thanks for the quick updates!!

**Review by ktabors — APPROVED** [review]

**Pull request merged**